jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py

```diff
@@ -2,51 +2,122 @@
 __license__ = "Apache-2.0"
 
 import copy
-from typing import Dict
-
-from jina import requests, DocumentArray, Executor
+import functools
+from typing import Dict, Optional
 
+from jina import DocumentArray, Executor, requests
 from jina_commons import get_logger
 
 try:
     from jinahub.indexers.searcher.FaissSearcher import FaissSearcher
-except:
-    from jina_executors.indexers.searcher.FaissSearcher.faiss_searcher import FaissSearcher
+except:  # noqa: E722
+    from jina_executors.indexers.searcher.FaissSearcher.faiss_searcher import (
+        FaissSearcher,
+    )
 
 try:
-    from jinahub.indexers.storage.PostgreSQLStorage import (
-        PostgreSQLStorage,
-    )
-except:
-    from jina_executors.indexers.storage.PostgreSQLStorage import (
-        PostgreSQLStorage,
-    )
+    from jinahub.indexers.storage.PostgreSQLStorage import PostgreSQLStorage
+except:  # noqa: E722
+    from jina_executors.indexers.storage.PostgreSQLStorage import PostgreSQLStorage
 
 
 class FaissPostgresSearcher(Executor):
-    """A Compound Indexer made up of a FaissSearcher (for vectors) and a Postgres Indexer"""
+    """A Compound Indexer made up of a FaissSearcher (for vectors) and a Postgres
+    Indexer
+
+    :param dump_path: a path to a dump folder containing
+    the dump data obtained by calling jina_commons.dump_docs
+    :param use_dump_func: whether to use the dump
+     function of PostgreSQLStorage, when dump_path is not provided
+    """
 
     def __init__(
         self,
-        dump_path=None,
+        dump_path: Optional[str] = None,
+        use_dump_func: bool = False,
+        total_shards: int = None,
```

Comment:
Either this becomes `Optional[int]` or needs another default. I prefer having default `= 1`.

Reply:
This needs to be passed from core, I suggest. I will make it optional and leave the warning there to stand out until solved.

Reply:
can we assume the total number of shards stays the same? I think in the short-term yes, but long-term we might want to scale the number of shards up and down

Reply:
Stays the same for all systems, or what do you mean? By default it's `None` unless you pass it.

Reply:
This total number of shards should be determined at the startup and never be `None`. And if it is None, it should default to `1` or so. This will make all downstream code easier (e.g. the `             if self.total_shards: ` below.